apple/internal/partials/swift_dynamic_framework.bzl

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Partial implementation for Swift dynamic frameworks."""
+
+load(
+    "@build_bazel_rules_apple//apple/internal:bundling_support.bzl",
+    "bundling_support",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:intermediates.bzl",
+    "intermediates",
+)
+load(
+    "@build_bazel_rules_apple//apple/internal:processor.bzl",
+    "processor",
+)
+load(
+    "@bazel_skylib//lib:partial.bzl",
+    "partial",
+)
+load(
+    "@bazel_skylib//lib:paths.bzl",
+    "paths",
+)
+
+def _get_header_imports(framework_imports):
+    """Get the header files from the list of framework imports"""
+
+    return [file for file in framework_imports if file.short_path.endswith(".h")]
+
+def _swift_dynamic_framework_partial_impl(ctx, swift_dynamic_framework_info):
```

Comment:
i think we're trying to move away from partials taking `ctx` as a param?

Reply:
I matched the format of the other partials and am no longer passing in `ctx` as a param to the rule when I call it. I added the same TODO that I saw in other partials to `Remove ctx from the args when ctx is removed from all partials`

When I removed it as a parameter I got `Error: _swift_dynamic_framework_partial_impl() got unexpected keyword argument: ctx` even though I am no longer manually passing in `ctx`.